### PR TITLE
Support for gradle flavors

### PR DESCRIPTION
**Pull request by piaskowyk**
## Description

Added support for Gradle flavor. Solution based on https://stackoverflow.com/a/34679933/7276742

Fixes
- https://github.com/software-mansion/react-native-reanimated/issues/2245
- https://github.com/software-mansion/react-native-reanimated/issues/2292

### CODE REVIEW

File: android-npm/build.gradle [hunk 1]
```diff
@@ -60,16 +82,16 @@ rootProject.getSubprojects().forEach({project ->
                 project.getProperties().android.packagingOptions.pickFirst("lib/**/libfbjni.so")
 
                 replaceSoTaskDebug.dependsOn(
-                    project.getTasks().getByPath("${appName}:mergeDebugNativeLibs"),
-                    project.getTasks().getByPath("${appName}:stripDebugDebugSymbols")
+                    project.getTasks().getByPath("${appName}:merge${flavorString}DebugNativeLibs"),
```
Comment (Szymon20000) [inline android-npm/build.gradle:85]: What about build variant does it also has an impact on the task name?

Comment (Szymon20000) [reply android-npm/build.gradle:85]: I didn't understand a build variant term properly. It turned out that it's just a combination of build type and a flavor.

**Review by Szymon20000 — APPROVED** [review]

**Review by mikehardy — COMMENTED** [review]
Looks great @piaskowyk - thanks - sorry I couldn't PR it myself!

**Comment by piaskowyk** [comment]
@mikehardy your comments were really helpful! Great cooperation for me 😄

**Comment by ilyausorov** [comment]
@piaskowyk @Szymon20000 hey guys, I've been trying to get through this issue of installing react-native-reanimated@2.2.2 with RN 63.4 and multiple productFlavors configured and this solution doesn't seem to be returning the product flavor to properly finish the gradle sync process.

What I am seeing is that the value of `taskRequestName` in `getCurrentFlavor()` comes back as `[DefaultTaskExecutionRequest{args=[],projectPath='null'}]` and therefore the flavor returns as an empty string and the gradle sync process fails.

I've had to add a patch to hard code my product flavor (in each respective branch) so that it successfully builds 
`def flavorString = 'Dev' // getCurrentFlavor()`

See screenshot from Android Studio logs where I added a few println's:

image

Any ideas why I'm not getting any value for the flavor?

**Comment by mikehardy** [comment]
This isn't going to be the most helpful, but with react-native 0.65.1 I don't see a problem, it appears to work for me

**Comment by ilyausorov** [comment]
@mikehardy that’s correct. If you look at the code, the get flavor code only runs if the react native version is below 65

**Comment by dark1451** [comment]
@ilyausorov 

I had the same experience.

The link's answer succeeds in getting the correct build value.

If I directly modify gradle like this, it is not tracked in git, so it is unsettling.
Or need to downgrade the reanimated version or upgrade the react-native version...

I hope it merges into that solution.

react-native-cli: 2.0.1
react-native: 0.64.2

https://github.com/software-mansion/react-native-reanimated/issues/2292#issuecomment-901100248

**Comment by mikehardy** [comment]
@dark1451 

> If I directly modify gradle like this, it is not tracked in git, so it is unsettling.


Use patch-package: https://github.com/ds300/patch-package - 6,286 stars for a reason, it's a vital tool - cheers

**Pull request merged**